File: src/main/java/kr/co/bookand/backend/account/domain/Role.java
package kr.co.bookand.backend.account.domain;

import kr.co.bookand.backend.account.exception.AccountException;
import kr.co.bookand.backend.common.exception.ErrorCode;

public enum Role {
    USER, ADMIN, MANAGER, BAN;

    public void checkAdmin() {
        if (this != ADMIN) {
            throw new AccountException(ErrorCode.ROLE_ACCESS_ERROR, this);
        }
    }

    public void checkNotUser() {
        if (this != MANAGER && this != ADMIN) {
            throw new AccountException(ErrorCode.ROLE_ACCESS_ERROR, this);
        }
    }
}

File: src/main/kotlin/kr/co/bookand/backend/account/domain/Role.kt
package kr.co.bookand.backend.account.domain

enum class Role {
    USER, ADMIN, MANAGER, SUSPENDED;

    fun checkAdmin() {
        if (this != ADMIN) {
            throw RuntimeException("ErrorCode.ROLE_ACCESS_ERROR")
        }
    }

    fun checkAdminAndManager() {
        if (this != MANAGER && this != ADMIN) {
            throw RuntimeException("ErrorCode.ROLE_ACCESS_ERROR")
        }
    }
}